.github/workflows/ci-image.yml

```diff
@@ -0,0 +1,33 @@
+name: Build Dev Image CI
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - "*"
```

Comment:
Do we need this on every branch push?

Reply:
Why not?   This is significantly faster than the existing ci integration suite.  Also doesn't the fast feedback help if you have a commit with issues/warnings/fmt problems?   Is there any harm to that we should consider?

Reply:
As an aside, considering that we have existing code format issues committed on main, I suspect that running these checks more frequently is a good thing for the overall health of the code.

Reply:
One reason is that we seem to be pushing all images to the tag of LATEST, so commits on feature branches will override main.

Agree the checking formatting on all branches is a good thing, but we could easily add that into our existing lint CI workflow.

Reply:
@neekolas  and I discussed offline and clarified that we both prefer to have fewer GitHub workflows.   I will take a look at the lint workflow and attempt to merge that into the containerized build.   

Going forward we should be able to achieve having all fmt/lint/unit testing in the containerized build and all integration tests (mls) in its own workflow.   The goal would be eventually there are about 2 workflows.     We won't achieve that in this PR, but that is a good target for the future.